File: 20190605/412. Fizz Buzz.cpp
class Solution {
public:
    vector<string> fizzBuzz(int n) {
        
        vector<string> res;
        
        for(int i = 1, fuzz = 0, buzz = 0; i <= n; i++)
        {
            fuzz++;
            buzz++;
            if(fuzz == 3 && buzz == 5)
            {
                res.push_back("FizzBuzz");
                fuzz = 0;
                buzz = 0;
            }
            else if(fuzz == 3)
            {
                res.push_back("Fizz");
                fuzz = 0;
            }
            else if(buzz == 5)
            {
                res.push_back("Buzz");
                buzz = 0;
            }
            else
                res.push_back(to_string(i));
        }
        
        return res;

        
    }
};

File: 20190605/412. Fizz Buzz.py
class Solution:
    def fizzBuzz(self, n: int) -> List[str]:
        return ["Fizz" * (not i % 3) + "Buzz" * (not i % 5) or str(i) for i in range(1, n+1)]
        # or:找到第一个非空对象